src/packages/src/xroad/ubuntu/generic/xroad-center.config

```diff
@@ -3,7 +3,7 @@
 . /usr/share/debconf/confmodule
 
 get_prop() {
-  crudini --get "$1" '' "$2" 2>/dev/null || echo -n "$3"
+  awk -F "=" '/'"$2"'/ {print $2}' "$1" | tr -d ' '
```

Comment:
What was the reason for replacing `crudini` with `awk` here? `crudini` should be available, since it is a transitive pre-dependency of xroad-center. Also the new version has different behavior, it does not return the default ($3) in case it does not find anything (used on line 26).

Reply:
As you suspected, the reason was that `crudini` seems to be not available at the time when the code is executed. Even moving it to pre-depends section of xroad-base doesn't fix it. I think returning the default is not needed, so I removed it from the calls in https://github.com/nordic-institute/X-Road/pull/1024/commits/db63a2172197f094aa53dcd65a2e523d2df19fb2

Reply:
I tested this with clean install, and crudini was present when the configure script called it. Several scripts use crudini at preinst stage, so it would be odd that it does not work here. Are you sure that the root cause was missing crudini?

Reply:
Ok, I was able to reproduce the problem (requires that apt-utils package is installed). So we need to fix the xroad-proxy.config as well. awk is btw not "essential" either, so it might be possible that it is not present in some corner case.

Reply:
Awk reverted to crudini in branch https://github.com/nordic-institute/X-Road/tree/XRDDEV-1706-with-crudini
The installation log contains this:
```
Preconfiguring packages ...
testing crudini
/tmp/xroad-center.config.J8UJr8: line 25: crudini: command not found
xroad-center failed to preconfigure, with exit status 127
Selecting previously unselected package libjpeg-turbo8:amd64.
(Reading database ... 34904 files and directories currently installed.)
Preparing to unpack .../0-libjpeg-turbo8_1.5.2-0ubuntu5.18.04.4_amd64.deb ...
Unpacking libjpeg-turbo8:amd64 (1.5.2-0ubuntu5.18.04.4) ...
Selecting previously unselected package libmhash2:amd64.
Preparing to unpack .../1-libmhash2_0.9.9.9-7_amd64.deb ...
Unpacking libmhash2:amd64 (0.9.9.9-7) ...
Selecting previously unselected package libpython2.7-minimal:amd64.
Preparing to unpack .../2-libpython2.7-minimal_2.7.17-1~18.04ubuntu1.6_amd64.deb ...
Unpacking libpython2.7-minimal:amd64 (2.7.17-1~18.04ubuntu1.6) ...
Selecting previously unselected package python2.7-minimal.
Preparing to unpack .../3-python2.7-minimal_2.7.17-1~18.04ubuntu1.6_amd64.deb ...
Unpacking python2.7-minimal (2.7.17-1~18.04ubuntu1.6) ...
Selecting previously unselected package python-minimal.
Preparing to unpack .../4-python-minimal_2.7.15~rc1-1_amd64.deb ...
Unpacking python-minimal (2.7.15~rc1-1) ...
Selecting previously unselected package libpython2.7-stdlib:amd64.
Preparing to unpack .../5-libpython2.7-stdlib_2.7.17-1~18.04ubuntu1.6_amd64.deb ...
Unpacking libpython2.7-stdlib:amd64 (2.7.17-1~18.04ubuntu1.6) ...
Selecting previously unselected package python2.7.
Preparing to unpack .../6-python2.7_2.7.17-1~18.04ubuntu1.6_amd64.deb ...
Unpacking python2.7 (2.7.17-1~18.04ubuntu1.6) ...
Selecting previously unselected package libpython-stdlib:amd64.
Preparing to unpack .../7-libpython-stdlib_2.7.15~rc1-1_amd64.deb ...
Unpacking libpython-stdlib:amd64 (2.7.15~rc1-1) ...
Setting up libpython2.7-minimal:amd64 (2.7.17-1~18.04ubuntu1.6) ...
Setting up python2.7-minimal (2.7.17-1~18.04ubuntu1.6) ...
Linking and byte-compiling packages for runtime python2.7...
Setting up python-minimal (2.7.15~rc1-1) ...
Selecting previously unselected package python.
(Reading database ... 35666 files and directories currently installed.)
Preparing to unpack .../00-python_2.7.15~rc1-1_amd64.deb ...
Unpacking python (2.7.15~rc1-1) ...
Selecting previously unselected package rlwrap.
Preparing to unpack .../01-rlwrap_0.43-1_amd64.deb ...
Unpacking rlwrap (0.43-1) ...
Selecting previously unselected package python-six.
Preparing to unpack .../02-python-six_1.11.0-2_all.deb ...
Unpacking python-six (1.11.0-2) ...
Selecting previously unselected package python-iniparse.
Preparing to unpack .../03-python-iniparse_0.4-2.2_all.deb ...
Unpacking python-iniparse (0.4-2.2) ...
Selecting previously unselected package crudini.
Preparing to unpack .../04-crudini_0.7-1_amd64.deb ...
Unpacking crudini (0.7-1) ...
```
Later on, in the same installation it seems to be available:

```
Setting up xroad-center (7.0.0-0.20210820095542gitb3a0d16.ubuntu18.04) ...
Configuration file '/etc/xroad/db.properties'
 ==> File on system created by you or by a script.
 ==> File also in package provided by package maintainer.
   What would you like to do about it ?  Your options are:
    Y or I  : install the package maintainer's version
    N or O  : keep your currently-installed version
      D     : show the differences between the versions
      Z     : start a shell to examine the situation
 The default action is to keep your current version.
*** db.properties (Y/I/N/O/D/Z) [default=N] ? 
testing crudini
0
val=true

```

Reply:
yes, if `apt-utils` package is installed, the configure script is executed early (before even installing any of the dependencies pre-dependencies of any of the packages) --> crudini can not be used in config script.

Reply:
The same fix was done for the security server in commit https://github.com/nordic-institute/X-Road/pull/1024/commits/e613eb064321634a0f60cb01eb24b935ac87d018